components/Elevation/tests/unit/MaterialElevationColorTests.m

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialElevation.h"
+#import "MaterialMath.h"
+#import "UIColor+MaterialDynamic.h"
+
+@interface MaterialElevationColorTests : XCTestCase
+
+@end
+
+@implementation MaterialElevationColorTests
+
+- (void)testResolvedColorWithElevation {
```

Comment:
Please test the following use cases (at a minimum).  The final test conditions would probably be the cross-product of the different conditions. 

## Color Inputs
1. RGB Color space (the default)
2. Grey scale color space
3. P3Display color space
4. A pattern-based color

## Elevation inputs
1.  Elevation of 0
2. A positive elevation within reasonable values (*e.g.*, 0-24) [already done for RGB]
3. A positive elevation outside any reasonable value (*e.g.*, 10,000)
4. A negative elevation

Consider combining the inputs by testing multiple color objects in a single test. For example,

```objc
- (void)testResolvedColorWithNegativeElevation {
  // Given
  const CGFloat elevation = -10;
  // When
  UIColor *resolvedGrayScaleColor = [self.grayScaleColor resolvedColorWithElevation:elevation];
  UIColor *resolvedRGBColor = [self.rgbColor resolvedColorWithElevation:elevation];
  UIColor *resolvedP3DisplayColor = [self.p3DisplayColor resolvedColorWithElevation:elevation];
  UIColor *resolvedPatternColor = [self.patternColor resolvedColorWithElevation:elevation];

  // Then
  UIColor expectedGrayScaleColor = ...
  XCTAssertTrue([self compareColorsWithFloatPrecisionFirstColor:resolvedGrayScaleColor secondColor:expectedGrayScaleColor], @"(%@) is not equal to (%@)", resolvedGrayScaleColor, expectedGrayScaleColor);
  ...
}

Reply:
Addressed in ec78c1cf6e8932e4ac318e277dae7e31646bcab5